### DISCUSSION TITLE
误翻译部分位置的内容

### DISCUSSION THREAD

**Discussion by lingbopro** [maboloshi/github-chinese#302] [Ideas]
> [!NOTE]
> **后注：**
> 
> 我原本在这条里提到的这些内容在开发版都已解决，不过这个 issue 下面还有很多待解决的问题
> -- lingbopro, 2024-11-16

<details>
<summary>原内容</summary>

很不错的脚本，但是有时候会把不该翻译的地方（比如代码浏览）里的英文内容给改了

**我目前遇到了:** 代码浏览，文件树，diff浏览、Markdown浏览

### **以下为部分截图:**

<img width="872" alt="image" src="https://github.com/user-attachments/assets/46cc24e1-3c0d-4923-b7a7-1ff5ba31a622">

`resources`在文件树中被误替换成`资源`（<https://github.com/2dust/v2rayN/tree/master/v2rayN/v2rayN>）

-----

<img width="774" alt="image" src="https://github.com/user-attachments/assets/ed92d4b4-9f8f-4ac5-92f4-40eb5620e8c1">

`Pages`在文件树中被误替换成`GitHub Pages`（<https://github.com/AvaloniaCommunity/Material.Avalonia/tree/master/Material.Avalonia.Demo>）

-----

<img width="539" alt="image" src="https://github.com/user-attachments/assets/f51fdff5-25a1-474e-89ce-071d9cf012f1">

Markdown浏览中的文本被替换（<https://github.com/AvaloniaCommunity/Material.Avalonia/?tab=readme-ov-file#-overview>）

-----

<img width="530" alt="image" src="https://github.com/user-attachments/assets/9e20a334-f229-49f7-aeae-4655e18b2b75">

Wiki中的文本被替换（<https://github.com/AvaloniaCommunity/Material.Avalonia/wiki/Getting-started>）

-----

别的图找不着了

PS: 在GreasyFork上有一个关于这个问题的讨论：<https://greasyfork.org/zh-CN/scripts/435208/discussions/145988>

</details>

**Comment by maboloshi** [comment]
以上问题已在1.9.3的开发版中基本修复。欢迎测试

**Comment by lingbopro** [comment]
目前来看我给的这4个图在开发版中都修了，但是由于我忘了其它情况的复现链接了，所以我不到啊

**Comment by lingbopro** [comment]
又找到几个误翻译（已使用最新测试版）：

![image](https://github.com/user-attachments/assets/7af1de66-441a-4869-95c8-ed0a62eb3acb)

首页”探索仓库“的仓库名（原本为”Blog“）

<img width="796" alt="image" src="https://github.com/user-attachments/assets/c799e09f-7624-4049-9715-cf6e471a0a37">

查询字符串被翻译

**Comment by wyc-26** [comment]
大家好，我是 wuyuncheng-26 的小号
我的大号由于未知原因被 ban，导致我发的评论全部都被 GitHub 吞掉了（悲
![IMG_6359](https://github.com/user-attachments/assets/96b76895-2b0f-4885-b16b-aa4eb7597647)
导致我被封号的原因很有可能是因为 quote reply 了这个病毒链接（但我明明改成 xxx 了啊！！！
总之在官方解封我的账号之前，就先用小号与大家交流了

**Comment by lingbopro** [comment]
那条评论已经没了啊

**Comment by wyc-26** [comment]
> 那条评论已经没了啊

因为我在我这里把它删掉了，我是协作者，有权限去删 comment
你们看不到那条提示的，因为我号被 ban 了

**Comment by lingbopro** [comment]
虽然但是，一个陌生人扔给你一个链接啥也不说让你安装，是个人都不会下的吧

**Comment by wyc-26** [comment]
> 虽然但是，一个陌生人扔给你一个链接啥也不说让你安装，是个人都不会下的吧

我也没说我下载了啊（
我看网上说下载了之后运行会被盗号（悲
我这不是被盗，是因为 quote reply 了
![image](https://github.com/user-attachments/assets/115b8058-be5d-48a8-acea-c0f91e652b97)

**Comment by lingbopro** [comment]
> > 虽然但是，一个陌生人扔给你一个链接啥也不说让你安装，是个人都不会下的吧
> 
> 我也没说我下载了啊（ 我看网上说下载了之后运行会被盗号（悲  
> 我这不是被盗，是因为 quote reply 了

我知道你没下，我的意思是这东西应该没人会信

**Comment by TC999** [comment]
还是小心点，已经有不少人被封号了

**Comment by lingbopro** [comment]
所以说一旦有这个链接就会封？

**Comment by TC999** [comment]
> 所以说一旦有这个链接就会封？

看了下，是引用回复概率封号

**Comment by TC999** [comment]
“博客”那个，大写会误翻，小写不会
（词条在公共区域规则下，对应位置是左侧栏下方

**Comment by ChinaGodMan** [comment]
加个黑名单单词列表，不翻译他们

**Comment by lingbopro** [comment]
主要的问题还是翻译区域，这种明显不属于固定的UI

**Comment by maboloshi** [comment]
这2个都不太好处理 🤣

**Comment by xinyihl** [comment]
> resources在文件树中被误替换成资源（[2dust/v2rayN@master/v2rayN/v2rayN](https://github.com/2dust/v2rayN/tree/master/v2rayN/v2rayN?rgh-link-date=2024-08-28T06%3A11%3A52Z)）

这个实际上修复不完全，直接打开不会翻译，但是折叠再展开之后还是会翻译

**Comment by maboloshi** [comment]
> > resources在文件树中被误替换成资源（[2dust/v2rayN@master/v2rayN/v2rayN](https://github.com/2dust/v2rayN/tree/master/v2rayN/v2rayN?rgh-link-date=2024-08-28T06%3A11%3A52Z)）
> 
> 这个实际上修复不完全，直接打开不会翻译，但是折叠再展开之后还是会翻译

收到 我有空看一下

**Comment by lingbopro** [comment]
> > resources在文件树中被误替换成资源（[2dust/v2rayN@master/v2rayN/v2rayN](https://github.com/2dust/v2rayN/tree/master/v2rayN/v2rayN?rgh-link-date=2024-08-28T06%3A11%3A52Z)）
> 
> 这个实际上修复不完全，直接打开不会翻译，但是折叠再展开之后还是会翻译

确实是这样的

**Comment by lingbopro** [comment]
![image](https://github.com/user-attachments/assets/f7b3bd5b-3503-4752-9a05-da8a6656e106)

还有，复数问题（结尾多了个s）

**Comment by TC999** [comment]
> ![image](https://github.com/user-attachments/assets/f7b3bd5b-3503-4752-9a05-da8a6656e106)
> 
> 还有，复数问题（结尾多了个s）

这条是正则规则，在表达式结尾加上`s?`就能解决

**Comment by maboloshi** [comment]
> 这条是正则规则，在表达式结尾加上`s?`就能解决

是的 
```
[/(\d+) in progress check/, "$1 个正在运行的检查"],
```
->
```
[/(\d+) in progress checks?/, "$1 个正在运行的检查"],
```

**Comment by lingbopro** [comment]
![image](https://github.com/user-attachments/assets/3b481a8e-538c-41cd-a59b-899d41c08540)

![image](https://github.com/user-attachments/assets/ab925563-2d77-44e3-9924-334f84e680f4)

`started` -> `标星`，这谁写的typo（）

**Reply by TC999** [reply]
修了，更新到最新开发版就行

**Comment by wyc-26** [comment]
> ![image](https://private-user-images.githubusercontent.com/148617407/376003163-3b481a8e-538c-41cd-a59b-899d41c08540.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkyNDgxNzAsIm5iZiI6MTcyOTI0Nzg3MCwicGF0aCI6Ii8xNDg2MTc0MDcvMzc2MDAzMTYzLTNiNDgxYThlLTUzOGMtNDFjZC1hNTliLTg5OWQ0MWMwODU0MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxOFQxMDM3NTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kYjNhNzAxYTYzNWE5NmNiMjg4NTFmZmQ1ZDk1YzkyMDAwZDI3OTI3NTIzOWVlMjkxN2MzN2VjZjYzMDEyY2E3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.2pj4FHsxdGADPEX4L3wel__NT2Vrgz_Vnkws7YN5wlQ)
> 
> ![image](https://private-user-images.githubusercontent.com/148617407/376003176-ab925563-2d77-44e3-9924-334f84e680f4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkyNDgxNzAsIm5iZiI6MTcyOTI0Nzg3MCwicGF0aCI6Ii8xNDg2MTc0MDcvMzc2MDAzMTc2LWFiOTI1NTYzLTJkNzctNDRlMy05OTI0LTMzNGY4NGU2ODBmNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAxOFQxMDM3NTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00NjI5NjUxYTMzNzFhZjZhMGVkMThkNzYyMGRkMGM2OTZjNDQyZTJmOTYxZWUwZmJjNzRkNzkyODA2Y2NmMzA4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.3Am64LejkJ9knXLLAIqx9CaQbwwl6AKLbSZwQ7zzXp4)
> 
> `started` -> `标星`，这谁写的typo（）

草 确实
<img src="https://bbs-static.miyoushe.com/static/2023/05/23/a2fa805a0a782cdb8ab44ddaf8d5343d_962898009752203374.png" alt="阿姬倒地了04" width="75" >

**Comment by lingbopro** [comment]
[![image](https://github.com/user-attachments/assets/bee63a33-7a4d-4e88-b172-e749613ff6d4)](https://github.com/lingbopro/soberjs-vscode/actions/runs/11366444633/job/31616858613)

还有这个，改我Actions输出

**Reply by TC999** [reply]
最新开发版已修

**Comment by nageih** [comment]
补充：这里的[`on`](https://github.com/maboloshi/github-chinese/blob/gh-pages/locals.js#L1037)在英文单词中会被误翻译为`于`，甚至`json`也变成了`js于`，令人忍俊不禁
![image](https://github.com/user-attachments/assets/b48e19db-9e29-46f4-a93a-375d7c639b7b)

**Comment by lingbopro** [comment]
确实，有些在一个单词中间的也会被翻译

**Comment by lingbopro** [comment]
![image](https://github.com/user-attachments/assets/f4618439-73cb-4611-ac50-850d1d43144f)

还有这个，组织主页的仓库简介

<https://github.com/cygwin> 的最下面几行

**Comment by lingbopro** [comment]
这个，搜索结果

![image](https://github.com/user-attachments/assets/f8204bb5-9484-4e74-85bd-5d18dc55132e)

`@ApiStatus.NonExtendable` 被翻译成了 `@Api状态。NonExtendable`，令人忍俊不禁

**Comment by lingbopro** [comment]
![image](https://github.com/user-attachments/assets/578e9a60-6546-4bf6-9e10-94f683a9e011)

这个`和`经常被误翻译（）

**Comment by maboloshi** [comment]
建议关闭 正则规则`[/ and /, " 和 "],`

**Comment by JamieBox2020** [comment]
还有经常会看到README文档的位置误翻译，能否加个判断不要翻译README文档区域的内容？
复现链接：`https://github.com/jason5ng32/MyIP`
![issue2](https://github.com/user-attachments/assets/56643bd6-0f66-4f0b-b551-9e11b637e698)
<br />
<br />
![issue2_resolve](https://github.com/user-attachments/assets/a8fa30bf-9f78-4bd6-8eee-1a7870badd13)

**Comment by lingbopro** [comment]
> 还有经常会看到README文档的位置误翻译，能否加个判断不要翻译README文档区域的内容？ 复现链接：`https://github.com/jason5ng32/MyIP` ...

这个和我最开始的那个是一样的问题，更新到最新开发版即可

**Comment by maboloshi** [comment]
> 还有经常会看到README文档的位置误翻译，能否加个判断不要翻译README文档区域的内容？ 复现链接：`https://github.com/jason5ng32/MyIP` 

[【开发版】](https://github.com/maboloshi/github-chinese/raw/gh-pages/main.user.js)没这个问题

**Comment by maboloshi** [comment]
> 这个，搜索结果

发一下示例链接

**Comment by lingbopro** [comment]
> > 这个，搜索结果
> 
> 发一下示例链接

https://github.com/search?q=Fabric+API+org%3AFabricMC&type=code

**Comment by maboloshi** [comment]
> [![image](https://private-user-images.githubusercontent.com/148617407/378136868-bee63a33-7a4d-4e88-b172-e749613ff6d4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE3NTYwMDMsIm5iZiI6MTczMTc1NTcwMywicGF0aCI6Ii8xNDg2MTc0MDcvMzc4MTM2ODY4LWJlZTYzYTMzLTdhNGQtNGU4OC1iMTcyLWU3NDk2MTNmZjZkNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExNlQxMTE1MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNDhhNjEzZDM5ZTBhMDgzYjA5ZWM5NjgyZDUwNzI5ZTgwMDBmNzAxOGYyYjFjNWZiNTkzNjUzN2Y3MzdlNTc1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.nDb2MBQ6wuo8wdh_grKx5i2D_0QmCTxuen31h8XZqk0)](https://github.com/lingbopro/soberjs-vscode/actions/runs/11366444633/job/31616858613)
> 
> 还有这个，改我Actions输出

屏蔽后，右侧的时间也屏蔽 OK不
<img width="641" alt="image" src="https://github.com/user-attachments/assets/d6490628-6865-4dc4-b727-810fd02c52e2">

是这个规则导致翻译
repository/actions:
`[/Download ([^ ]+)/, "下载 $1"],`

**Comment by maboloshi** [comment]
> > > 这个，搜索结果
> > 
> > 
> > 发一下示例链接
> 
> https://github.com/search?q=Fabric+API+org%3AFabricMC&type=code

这个貌似已经修复了

**Comment by maboloshi** [comment]
@TC999 hello 后续添加正则的需要在静态规则中同时添加出处, 便于后期排查
另外, 您这边已添加的正则补一下出处

**Comment by lingbopro** [comment]
> > > > 这个，搜索结果
> > > 
> > > 
> > > 发一下示例链接
> > 
> > 
> > https://github.com/search?q=Fabric+API+org%3AFabricMC&type=code
> 
> 这个貌似已经修复了

但是我用的（应该）是最新版，也有这个问题

**Comment by maboloshi** [comment]
> 但是我用的（应该）是最新版，也有这个问题

![image](https://github.com/user-attachments/assets/e3b08a4d-f997-4937-8c25-abb0bdc54659)

**Comment by TC999** [comment]
> > [![image](https://private-user-images.githubusercontent.com/148617407/378136868-bee63a33-7a4d-4e88-b172-e749613ff6d4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE3NTYwMDMsIm5iZiI6MTczMTc1NTcwMywicGF0aCI6Ii8xNDg2MTc0MDcvMzc4MTM2ODY4LWJlZTYzYTMzLTdhNGQtNGU4OC1iMTcyLWU3NDk2MTNmZjZkNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExNlQxMTE1MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNDhhNjEzZDM5ZTBhMDgzYjA5ZWM5NjgyZDUwNzI5ZTgwMDBmNzAxOGYyYjFjNWZiNTkzNjUzN2Y3MzdlNTc1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.nDb2MBQ6wuo8wdh_grKx5i2D_0QmCTxuen31h8XZqk0)](https://github.com/lingbopro/soberjs-vscode/actions/runs/11366444633/job/31616858613)
> > 还有这个，改我Actions输出
> 
> 屏蔽后，右侧的时间也屏蔽 OK不 <img alt="image" width="641" src="https://private-user-images.githubusercontent.com/7850715/386840583-d6490628-6865-4dc4-b727-810fd02c52e2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE4MTM1MjksIm5iZiI6MTczMTgxMzIyOSwicGF0aCI6Ii83ODUwNzE1LzM4Njg0MDU4My1kNjQ5MDYyOC02ODY1LTRkYzQtYjcyNy04MTBmZDAyYzUyZTIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTdUMDMxMzQ5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NmY5ZDMzNGZmNGUwZDNhN2YzNmMwYmFkMWU1YTRhOWQ5ZTY0ZjIxYjJjNmIzOTAxN2RkZDU3OTc2MDFkYzIyMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.u6BQOobGeg_yB_wvtoV_mjs8wZ3NAI6jvfa7WyBVkZo">
> 
> 是这个规则导致翻译 repository/actions: `[/Download ([^ ]+)/, "下载 $1"],`

这条规则对应的是这个位置：
![image](https://github.com/user-attachments/assets/6a2332d3-7f6e-4c1b-b0c6-cdcda2ecdd7e)

**Comment by TC999** [comment]
> [![image](https://private-user-images.githubusercontent.com/148617407/378136868-bee63a33-7a4d-4e88-b172-e749613ff6d4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE4MTQ4MTYsIm5iZiI6MTczMTgxNDUxNiwicGF0aCI6Ii8xNDg2MTc0MDcvMzc4MTM2ODY4LWJlZTYzYTMzLTdhNGQtNGU4OC1iMTcyLWU3NDk2MTNmZjZkNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExN1QwMzM1MTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZWVkNDAxN2Y2NWJhZWY1YjZiNTJlZmUwNGQ3M2ZjYTQ0ZTI4YTE3ZDk3ODY3NWJlZDNlNTg1YmVjYzMyNzRlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.RLVoiWwv4PdVUOwVTOlG1xY35kyflOAdQCUxhOloYwE)](https://github.com/lingbopro/soberjs-vscode/actions/runs/11366444633/job/31616858613)
> 
> 还有这个，改我Actions输出

![image](https://github.com/user-attachments/assets/cda599d3-38a0-4b5a-9a0d-204ba9a3f173)
解决
但里面的详细信息不打算处理（之前为部分输出日志加过词条（

**Comment by TC999** [comment]
建议把这个议题转为讨论，这样方便管理

**Comment by TC999** [comment]
最好把多余的评论或已解决的BUG隐藏一下
